src/components/SearchBar.tsx

```diff
@@ -16,11 +16,32 @@ export interface SearchBarState {
     detailed: boolean;
 }
 
+interface SearchDisplayModeChangedFunc {
+    (value: boolean): void;
+}
+
+interface SearchCategoriesChangedFunc {
+    (categories: string[]): void;
+}
+
+interface SearchTextChangedFunc {
+    (event: any): void;
+}
+
```

Comment:
By right function prototypes should come before `export interface SearchBarProps`.